### PR TITLE
Converted test projects to run on netcoreapp2.0

### PR DESCRIPTION
**Pull request by kichalla**

### CODE REVIEW

File: test/Microsoft.AspNetCore.Server.KestrelTests/FrameResponseStreamTests.cs [hunk 1]
```diff
@@ -61,6 +61,9 @@ public void BeginReadThrows()
             var stream = new FrameResponseStream(new MockFrameControl());
             Assert.Throws<NotSupportedException>(() => stream.BeginRead(new byte[1], 0, 1, null, null));
         }
+#elif NETCOREAPP2_0
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.Server.KestrelTests/FrameResponseStreamTests.cs]: ?

Comment (kichalla) [inline test/Microsoft.AspNetCore.Server.KestrelTests/FrameResponseStreamTests.cs]: This is just a safe guard to fail compilation when TFMs have been changed in the .csproj but not here.

**Comment by pranavkm** [comment]
The build is failing.

**Comment by pranavkm** [comment]
Still failing.

**Comment by davidfowl** [comment]
I'll take a look later.

File: test/Microsoft.AspNetCore.Server.KestrelTests/FrameResponseStreamTests.cs (hunk 1 above)
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.KestrelTests/FrameResponseStreamTests.cs]: Please add a comment `// Safe guard to fail compilation when TFMs have been changed in the .csproj but not in code

Easy to forget about this in the future and remove it.

File: test/Microsoft.AspNetCore.Server.Kestrel.Performance/ResponseHeadersWritingBenchmark.cs [hunk 2]
```diff
@@ -38,7 +37,7 @@ public class ResponseHeadersWritingBenchmark
         {
             _frame.Reset();
             _frame.StatusCode = 200;
-            _frame.HttpVersionEnum = HttpVersion.Http11;
+            _frame.HttpVersionEnum = Internal.Http.HttpVersion.Http11;
```
Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.Performance/ResponseHeadersWritingBenchmark.cs]: ?

File: test/Microsoft.AspNetCore.Server.KestrelTests/FrameResponseStreamTests.cs (hunk 1 above)
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Server.KestrelTests/FrameResponseStreamTests.cs]: We're standardizing across this in all the projects. I think it'll be overkill to specify this eveywhere we use this.

**Comment by davidfowl** [comment]
🆙 📅 

/cc @pranavkm @kichalla

**Comment by davidfowl** [comment]
Docker is failing. I think we need to use a new base image for:
https://github.com/aspnet/KestrelHttpServer/blob/kichalla/netcoreapp20/test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/SystemdActivation/Dockerfile

**Comment by davidfowl** [comment]
Finished this here https://github.com/aspnet/KestrelHttpServer/pull/1576

**Pull request closed without merging**